### PR TITLE
Fix mean computation for the geometric distribution in the data generator

### PR DESCRIPTION
**Pull request by vuule**
## Description
Since we moved random data generation to the GPU, geometric distribution has been approximated by half-normal distribution. However, the mean computation wasn't updated, causing a ~20% higher mean that the actual generated values.
Another issue that exasperated the problem is the implicit conversion to ints in the random generator. This effectively lowered the mean of generated values by 0.5.

Together, these lead to list columns having the last row with more than 20% of the total column data. Huge single row caused low performance in many benchmarks. For example, Parquet files end up with a few huge pages and load imbalance in decode.

This PR fixes the mean computation to reflex the actual distribution, and rounds the random values when converting to ints. The result is a correct distribution of the number of elements in each randomly generated list.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by vuule** [comment]
CC @shrshi

**Comment by vuule** [comment]
Parquet reader benchmarks with only list columns are faster by 50%, while the ones with mixed types are 20-30X faster.
As expected, there's no impact on performance measurements with other types. The file size has not changed significantly either.

File: cpp/benchmarks/common/generate_input.cu [hunk 1]
```diff
@@ -77,14 +77,14 @@ double get_distribution_mean(distribution_params<T> const& dist)
...
     case distribution_id::UNIFORM: return (dist.lower_bound / 2.) + (dist.upper_bound / 2.);
     case distribution_id::GEOMETRIC: {
-      auto const range_size = dist.lower_bound < dist.upper_bound
-                                ? dist.upper_bound - dist.lower_bound
-                                : dist.lower_bound - dist.upper_bound;
-      auto const p          = geometric_dist_p(range_size);
+      // In the current implementation, the geometric distribution is
+      // approximated by absolute value of a uniform distribution
+      auto const gauss_std_dev   = geometric_as_gauss_std_dev(dist.lower_bound, dist.upper_bound);
+      auto const half_gauss_mean = gauss_std_dev * sqrt(2. / M_PI);
```
Comment (vuule) [inline cpp/benchmarks/common/generate_input.cu:83]: from https://en.wikipedia.org/wiki/Half-normal_distribution :grin:

**Review by PointKernel — APPROVED** [review]

**Comment by vuule** [comment]
@PointKernel made some additional changes; requested your review so I don't sneak them past you :)

**Review by PointKernel — APPROVED** [review]

**Review by pmattione-nvidia — APPROVED** [review]

File: cpp/benchmarks/common/random_distribution_factory.cuh [hunk 2]
```diff
@@ -89,10 +91,17 @@ class geometric_distribution : public thrust::random::normal_distribution<integr
   T _lower_bound;
   T _upper_bound;
 
+  super_t make_appox_normal_dist(T lower_bound, T upper_bound) const
```
Comment (shrshi) [inline cpp/benchmarks/common/random_distribution_factory.cuh:94]: nit: Is it `make_approx_normal_dist`?

File: cpp/benchmarks/common/generate_input.cu [hunk 3]
```diff
@@ -77,14 +77,15 @@ double get_distribution_mean(distribution_params<T> const& dist)
     case distribution_id::NORMAL:
     case distribution_id::UNIFORM: return (dist.lower_bound / 2.) + (dist.upper_bound / 2.);
     case distribution_id::GEOMETRIC: {
-      auto const range_size = dist.lower_bound < dist.upper_bound
-                                ? dist.upper_bound - dist.lower_bound
-                                : dist.lower_bound - dist.upper_bound;
-      auto const p          = geometric_dist_p(range_size);
+      // Geometric distribution is approximated by a half-normal distribution
```
Comment (shrshi) [inline cpp/benchmarks/common/generate_input.cu:80]: :fire:

**Review by shrshi — APPROVED** [review]
Just one nit, otherwise looks great to me! Thank you :)

**Pull request merged**